pkg/cli/tail_test.go

```diff
@@ -230,3 +420,20 @@ func (p *fakePuller) Pull(ctx context.Context, filter string, maxNum int) ([]*lo
 	p.gotMaxNum = maxNum
 	return p.logEntries, p.injectErr
 }
+
+func (p *fakePuller) StreamPull(ctx context.Context, filter string, logCh chan<- *loggingpb.LogEntry) error {
+	p.gotFilter = filter
+	if p.injectErr != nil {
+		return p.injectErr
+	}
+	for _, v := range p.logEntries {
+		logCh <- v
+	}
+
+	// mock real streamPull's behavior, looping until canceled instead of using
+	// a for{} that loops forever, this is more efficient in test in
+	// TestStreamTailCommand, the contex time out is set to 2 seconds sleep 3
+	// seconds here make sure the server is still running before the time out.
+	time.Sleep(3 * time.Second)
+	return nil
```

Comment:
Using sleep is usually frowned upon. We should avoid it if we can.

Maybe try:

```suggestion
	for _, v := range p.logEntries {
		logCh <- v
	}

	<-ctx.Done()
	return nil
```